tools/contributors.mjs

```diff
@@ -0,0 +1,113 @@
+#!/usr/bin/env node
+import { execSync } from 'child_process'
+import { readFileSync, writeFileSync } from 'fs'
+import process from 'process'
+
+import { Octokit } from '@octokit/core'
+import { paginateRest } from '@octokit/plugin-paginate-rest'
+
+/**
+ * Parses a contributor string into it's parts
+ * @param {string} entry
+ * @returns {name: string, email?: string, web?: string}
+ */
+const parseContributorString = (entry) => {
+  let name, email, web
+
+  const matchFull = /^(.+)\s<(.+)>\s\((.+)\)$/gm.exec(entry)
+  const matchWeb = /^(.+)\s\((.+)\)$/gm.exec(entry)
+  const matchMail = /^(.+)\s<(.+)>$/gm.exec(entry)
+  if (matchFull) {
+    ;[, name, email, web] = matchFull
+  } else if (matchWeb) {
+    ;[, name, web] = matchWeb
+  } else if (matchMail) {
+    ;[, name, email] = matchMail
+  } else {
+    name = entry
+  }
+
+  return { name, email, web }
+}
+
+/**
+ * Generates a contributor string out of an entry
+ * @param {object} entry
+ * @param {string} entry.name
+ * @param {string} [entry.email]
+ * @param {string} [entry.web]
+ * @returns
+ */
+const createContributorString = (entry) =>
+  [entry.name, entry.email && `<${entry.email}>`, entry.web && `(${entry.web})`].filter(Boolean).join(' ')
+
+// read the packageJSON
+const packageJson = JSON.parse(readFileSync('package.json', 'utf-8'))
+// parse the existing contributors
+const existingContributors = packageJson.contributors.map((contributor) => parseContributorString(contributor))
+
+// Get a list of email addresses from local git log as they are not
+// part of the user information
+const mailList = new Map(
+  execSync(`git log --format='%an⏣%ae'`)
+    .toString()
+    .split('\n')
+    .map((entry) => entry.split('⏣'))
+    .filter(([key]) => !(key.length === 0 || key.includes('[bot]'))),
+)
+
+const [GITHUB_TOKEN] = process.argv.slice(2)
+
+if (!GITHUB_TOKEN) {
+  throw new Error('Please provide the GITHUB_TOKEN as argument to the command: node ./tools/contributors.mjs <token>')
+}
+
+const PagedOctokit = Octokit.plugin(paginateRest)
+const octokit = new PagedOctokit({ auth: GITHUB_TOKEN })
+
+const contributorList = await octokit.paginate('GET /repos/{owner}/{repo}/contributors', {
```

Comment:
This will give us the existing contributors list, so I'm not sure if it includes first time contributors.
That is, if a first time user opens a PR, their name won't be added in that PR, but in the next on (I think), which can be confusing.

We probably want to look at the `git log` for the current commit, and use that as a reference

Reply:
Maybe a better way will be to take the PR author from the GitHub action event and use that one